src/config_design.config

```diff
@@ -0,0 +1,126 @@
+
+%% ----------------------------------------------------------------------------
+%% WORKER_GROUP SUPPORT: https://github.com/cloudant/perf/issues/32
+%% ----------------------------------------------------------------------------
+%%
+%% "worker_types" would provide per-worker config list,
+%%     miniature copy of current config file for what is different from global
+
+{worker_types, [
+   %% Each worker_type can have its own list of config options
+   %%     to be used on top of global parameters known to basho_bench_config
+   {worker_name_a, [ {weight, 5}, 
+                     {driver, cb_driver}, 
+                     {operations, [
+                         {get,1},
+                         {put, [ {count,1}, 
+                                 {key_generator, keyGen1}
+                               ]}
+                         ]}, 
+                     {other, 1}
+                   ]},
+   {worker_name_b, []}
+   ]}.
+
+{workers, [
+   {worker_name_a, 1},
+   {worker_name_b, 3}
+   ]}.
+
+%% IMPLEMENTATION:
+%% The "concurrent" property picks how many workers but use workers weights for proportions needed. 
+%%
+%% basho_bench_worker_config:next_worker 
+%%    pops worker_name off weighted list based on "workers" proportions
+%%
+%% basho_bench_worker:init 
+%%    if next_worker is defined then 
+%%        looks up worker type to get per-worker-type config as LocalConfig
+%%        uses config_get(Key, LocalConfig) for initialization
+%%        stores LocalConfig into state for future use
+%%        benchmark applications can use config_get and then save values into own driverState
+%% 
+%% basho_bench_worker:config_get(Key,Default,LocalConfig) 
+%%    checks LocalConfig first then uses basho_bench_config:get for global defaults
+%%
+%% ISSUES
+%% 1. Some parms (source_dir, pre/post_hook, etc) and funs like log_dimensions may need more thought
+%%    to handle worker_types since these are currently handled outside basho_bench_worker.
+%%    May end up with list of pre- and post-hooks to run across multiple drivers, etc. 
+
+
+%% ----------------------------------------------------------------------------
+%% SETUP/TEARDOWN: https://github.com/cloudant/perf/issues/35
+%% ----------------------------------------------------------------------------
+%%
+%% Existing code already tries to call a "terminate" function
+%%    which is not documented or used in existing drivers
+%%
+%%     %% Give the driver a chance to cleanup
+%%    (catch (State#state.driver):terminate({'EXIT', Reason}, State#state.driver_state)),
+%%
+%% Option 1: Make automatic calls to driver:setup and driver:terminate using (catch)
+%% 
+%% Option 2: Add new params to explicitly identify functions, like driver pre/post_hook work now
+
+{worker_setup, SetupFunctionName}.  %% Call at end of basho_bench_worker:init with State
+
+{worker_teardown, TearDownFunctionName}.  %% leverage existing call to State.driver:terminate
```

Comment:
I experimented with terminate call and that seems to work fine, getting called after a SUCCESSFUL run when the thread is being stopped. Need to discuss what requirements really are here. 

terminate(Status, State) ->
    T = basho_bench_worker:config_get(test_value),
    io:format("NEW: TEST PROCESS ENV config_get (~p = ~p) Status ~p~n   State ~p~n", [test_value, T, Status, State]).